HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterECALUncorrectedL1Seeded_cfi.py

```diff
@@ -111,6 +111,7 @@
     ),
     recHitCleaners = cms.VPSet(),
     recHitsSource = cms.InputTag("hltParticleFlowRecHitECALL1Seeded"),
+    usePFThresholdsFromDB = cms.bool(False), 
```

Comment:
any reason to keep the phase-2 HLT behind offline at this point?

Reply:
i'm not sure I understood the question. For ECAL, this boolean needs to be False as ECAL thresholds are handled differently. In offline config also, for ECAL it is false.

Reply:
It's confusing to have the same parameter name if different thresholds are applied.

Reply:
The same code runs for all calorimeters (ECAL, HB, HE, HO, HF). So it is kept as True for HB and HE, where it is needed. And for others it is kept as False.

Reply:
I think changing this behaviour would profit the maintainability of the code.

Reply:
Right now I don't see any easy way to do it. It can take some thoughts and some time. 
It might be useful to understand which of the suggestions in this PR are showstopper to sign it, and which are more for general improvement of code and its maintenance which can be addressed later. This way we can plan/prioritize our tasks accordingly.

Reply:
> The same code runs for all calorimeters (ECAL, HB, HE, HO, HF). So it is kept as True for HB and HE, where it is needed. And for others it is kept as False.

@swagata87 I think I still need some clarification on this point. 
For example in the case of this file you are touching the configuration of `PFClusterProducer`. 
Looking at the code here

https://github.com/cms-sw/cmssw/blob/2fbc415053a78a9c688967b8ed1d85bc4574374a/RecoParticleFlow/PFClusterProducer/plugins/PFClusterProducer.cc#L63C41-L63C62

if `usePFThresholdsFromDB` is `False`, then `cutsFromDB` is `False` and the data member `paramPF` remains a null-pointer. Now `paramPF` is a pointer to `HcalPFCuts`. Can you explain how having a non-null `HcalPFCuts` object influences the clustering in the other calorimeters? Thank you.

Reply:
If we set `usePFThresholdsFromDB` as True for other calorimeters (ECAL, PS, HO, HF), then physics-wise there should not be any change in results. Because, say the `PFClusterProducer` calls the `LocalMaximumSeedFinder::findSeeds` with non-null  `HcalPFCuts` for ECAL. This is not a problem because the `LocalMaximumSeedFinder::findSeeds` uses `HcalPFCuts` when the PFLayer is HB or HE --> https://github.com/cms-sw/cmssw/blob/2fbc415053a78a9c688967b8ed1d85bc4574374a/RecoParticleFlow/PFClusterProducer/plugins/LocalMaximumSeedFinder.cc#L132-L137

same for other codes, where for ECAL, PS, HO, HF, there is no usage of the non-null pointer 
https://github.com/cms-sw/cmssw/blob/2fbc415053a78a9c688967b8ed1d85bc4574374a/RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowClusterizer.cc#L238-L242

But I thought its better to keep `usePFThresholdsFromDB` as false for ECAL etc, so that we will not even read the HCAL thresholds from GT in the beginRun of `PFClusterProducer` if the clustering is happening in ECAL etc. So we will spend less time/resourse hopefully by keeping it False.

Reply:
I don't mean to delay this PR, but just a question for my own understanding. (the question is not specific to Phase 2).

In the case of ECAL (and following the analogy with HCAL), are any of the PFCluster config parameters (e.g. `seedingThreshold` or `gatheringThreshold`) related to the values in the GT record of type `EcalPFRecHitThresholdsRcd` ? Is the `EcalPFRecHitThresholds` tag used in PFlow producers at any point ? (if not, should that happen eventually ?)

Reply:
For ECAL, things work a bit differently. In the step where PFRecHits are produced from ECAL RecHits, we use the EcalPFRecHitThresholdsRcd from GT. These are per-crystal noise thresholds, so they are so many in numbers (~75K) that one has to use GT-based approach. Also, they are updated once a year, generally.
Here is the code for that:
https://github.com/cms-sw/cmssw/blob/d59e80b6862b73d0497919044d19841844f164b3/RecoParticleFlow/PFClusterProducer/interface/PFRecHitQTests.h#L53-L87

Then, in the clustering step, ECAL has seeding thresholds and gathering thresholds that are passed from the config file, here: https://github.com/cms-sw/cmssw/blob/786ae10dbb66c296c19ebd1f3c63bbc4a4d3ff58/RecoParticleFlow/PFClusterProducer/python/particleFlowClusterECALUncorrected_cfi.py#L51-L76
Since these are just one number for barrel and one for endcap, and they don't need update every year, there is no need to put those in GT. One can of course double check with ECAL DPG if they plan to update these numbers in Run3 or if they plan to put them in GT, but I don't think so.

Reply:
@swagata87 , thanks for the explanation !

Reply:
> But I thought its better to keep usePFThresholdsFromDB as false for ECAL etc, so that we will not even read the HCAL thresholds from GT in the beginRun of PFClusterProducer if the clustering is happening in ECAL etc. So we will spend less time/resourse hopefully by keeping it False.

I think the cost of this is pretty negligible. 
Anyway I am about to sign this PR, but I still find confusing to have the same parameters configured differently in the phase-2 HLT menu depending on the "flavour". Please consider adding a comment to each instance for better documentation.

Reply:
okay I have now added a comment to every place where it is False in Phase2 HLT configs.